### PR TITLE
Create info windows

### PR DESCRIPTION
**Pull request by MikolJiron**
Add info windows to each marker on the map and make them dynamically render only when the user clicks on them.

### CODE REVIEW

File: portfolio/src/main/webapp/map.js [hunk 1]
```diff
@@ -14,9 +14,29 @@
 
 class Map {
   constructor() {
+    // Information for each location on the map.
     this.dukeLocation = {lat: 36.001111, lng: -78.938889};
     this.wilsonLocation = {lat: 35.9974, lng: -78.9414};
     this.chapelLocation = {lat: 36.0019, lng: -78.9403};
+    this.keohaneLocation = {lat: 35.9989, lng: -78.9373}
+    this.dukeDescription = '<p class=info-window-description> \
```
Comment (hzimz) [inline portfolio/src/main/webapp/map.js:22]: This is a lot of html to keep in js, break up into html templates.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template

Comment (tinagrasso) [reply portfolio/src/main/webapp/map.js:22]: +1

File: portfolio/src/main/webapp/map.js [hunk 2]
```diff
@@ -14,9 +14,29 @@
...
+    this.chapelDescription = '<p class=info-window-description> \
+      This is Duke Chapel, one of the tallest university chapels in \
+      the world. It\'s architecture is Collegiate Gothic, many of Duke\'s \
+      academic buildings have this style of architecture. It\'s absolutely \
+      breathtaking.</p>';
+    this.keohaneDescription = '<p class=info-window-description> \
+      This is Keohane 4E, my dorm sophomore year. It is in a great location \
+      far enough away from main quad without being too far to be \
+      incovenient to walk to class. \
+      It\'s also one of the nicest dorms on campus.</p>';
```
Comment (tinagrasso) [inline portfolio/src/main/webapp/map.js:39]: You should create a <string, object> map to hold this data.

```
const locations = { 'duke': new Location(coords, descriptions), }
```

File: portfolio/src/main/webapp/map.js [hunk 3]
```diff
@@ -12,46 +12,101 @@
...
   constructor() {
+    // Zoom level for the map.
     this.ZOOM = 16;
-    this.MAP_COORDINATES = {
-      duke : {lat: 36.001111, lng: -78.938889},
-      wilson : {lat: 35.9974, lng: -78.9414},
-      chapel : {lat: 36.0019, lng: -78.9403}
-    };
+    // Map that contains Location objects representing each marker.
+    this.LOCATIONS = {
```
Comment (tinagrasso) [inline portfolio/src/main/webapp/map.js:25]: Move constants outside the constructor.

File: portfolio/src/main/webapp/map.js [hunk 4]
```diff
@@ -12,46 +12,101 @@
...
+      this.LOCATIONS.duke.description);
+
+    this.createMarker(this.LOCATIONS.wilson.coords, map,
+      this.LOCATIONS.wilson.title, this.LOCATIONS.wilson.description);
+
+    this.createMarker(this.LOCATIONS.chapel.coords, map,
+      this.LOCATIONS.chapel.title, this.LOCATIONS.chapel.description);
+
+    this.createMarker(this.LOCATIONS.keohane.coords, map,
+      this.LOCATIONS.keohane.title, this.LOCATIONS.keohane.description);
```
Comment (tinagrasso) [inline portfolio/src/main/webapp/map.js:74]: Replace with a foreach loop.

File: portfolio/src/main/webapp/map.js [hunk 5]
```diff
@@ -23,9 +23,23 @@ const DEFAULT_ZOOM = 16;
 class Map {
   constructor() {
     this.LOCATIONS = [
```
Comment (tinagrasso) [inline portfolio/src/main/webapp/map.js:25]: Please move this outside of the constructor.

**Pull request merged**